applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -221,20 +223,44 @@ static void cap_start_worker(struct k_work *work)
 	param.count = 0;
 	param.type = BT_CAP_SET_TYPE_AD_HOC;
 
-	if (unicast_server->sink_ep) {
-		cap_stream_params[param.count].member.member = unicast_server->device_conn;
-		cap_stream_params[param.count].stream = &unicast_server->cap_sink_stream;
-		cap_stream_params[param.count].ep = unicast_server->sink_ep;
-		cap_stream_params[param.count].codec_cfg = &lc3_preset_sink.codec_cfg;
-		param.count++;
+	/* Start all unicast_servers with valid endpoints */
+	for (int i = 0; i < ARRAY_SIZE(unicast_servers[idx.lvl1][idx.lvl2]); i++) {
+		unicast_server = &unicast_servers[idx.lvl1][idx.lvl2][i];
+		uint8_t state;
+
+		if (unicast_server->sink_ep) {
+			ret = le_audio_ep_state_get(unicast_server->sink_ep, &state);
+			if (state == BT_BAP_EP_STATE_STREAMING) {
+				LOG_DBG("Sink endpoint is already streaming, skipping start");
+				continue;
+			}
+
+			cap_stream_params[param.count].member.member = unicast_server->device_conn;
+			cap_stream_params[param.count].stream = &unicast_server->cap_sink_stream;
+			cap_stream_params[param.count].ep = unicast_server->sink_ep;
+			cap_stream_params[param.count].codec_cfg = &unicast_server->sink_codec_cfg;
+			param.count++;
+		}
+
+		if (unicast_server->source_ep) {
+			ret = le_audio_ep_state_get(unicast_server->source_ep, &state);
+			if (state == BT_BAP_EP_STATE_STREAMING) {
+				LOG_DBG("Source endpoint is already streaming, skipping start");
+				continue;
+			}
+
+			cap_stream_params[param.count].member.member = unicast_server->device_conn;
+			cap_stream_params[param.count].stream = &unicast_server->cap_source_stream;
+			cap_stream_params[param.count].ep = unicast_server->source_ep;
+			cap_stream_params[param.count].codec_cfg = &lc3_preset_source.codec_cfg;
+			param.count++;
+		}
 	}
 
-	if (unicast_server->source_ep) {
-		cap_stream_params[param.count].member.member = unicast_server->device_conn;
-		cap_stream_params[param.count].stream = &unicast_server->cap_source_stream;
-		cap_stream_params[param.count].ep = unicast_server->source_ep;
-		cap_stream_params[param.count].codec_cfg = &lc3_preset_source.codec_cfg;
-		param.count++;
+	if (param.count == 0) {
+		LOG_DBG("No streams to start for device %d.%d.%d", idx.lvl1, idx.lvl2, idx.lvl3);
```

Comment:
lvl2 is not applicable for unicast. Should we omit this in the print?

Reply:
Would be nice to get the entire index, in case idx2 for some reason is not 0